dvc/repo/ls.py

```diff
@@ -0,0 +1,109 @@
+import os
+
+from dvc.exceptions import NotDvcRepoError
+
+
+@staticmethod
+def ls(url, targets=[], rev=None, recursive=None):
+    if os.path.exists(url):
+        return _ls_local_repo(url, targets, recursive)
+
+    return _ls_git_repo(url, targets, rev, recursive)
+
+
+def _ls_files_local_repo(path, targets=[], recursive=None):
+    from dvc.repo import Repo
+
+    files = []
+    try:
+        repo = Repo(path)
+        if not len(targets):
+            targets.append(".")
+
+        def prettify_file(prefix, filename, is_root):
+            if not is_root:
+                filename = os.path.join(prefix, filename)
+            return filename
+
+        def prettify_dir(prefix, filename, is_root):
+            filepath = prettify_file(prefix, filename, is_root)
+            return filepath + "/"
+
+        for target in targets:
+            target_path = os.path.join(path, target)
+            if not os.path.exists(target_path):
+                continue
+
+            for dirpath, dirnames, filenames in repo.tree.walk(target_path):
+                prefix = os.path.relpath(dirpath, path)
+                is_root = os.path.abspath(dirpath) == os.path.abspath(path)
+                files.extend(
+                    map(lambda f: prettify_dir(prefix, f, is_root), dirnames)
+                )
+                files.extend(
+                    map(lambda f: prettify_file(prefix, f, is_root), filenames)
+                )
+                if not recursive:
+                    break
+
+    except NotDvcRepoError:
+        pass
+    finally:
+        repo.close()
+    return files
```

Comment:
Overall this looks broken:
```bash
$ tree .
├── a
│   ├── 1.txt
│   └── 2.txt
└── b
    ├── 1.txt
    └── 3.txt

$ dvc list .
1.txt
2.txt
3.txt
```
The output format should be different for cases of single and multiple targets.

Reply:
Can you please provide steps for reproduce?
```
(dvc) jiojiajiu@pon4ik:~/workspace/dvc/list (dvc_list)$ tree .
.
├── a
│   ├── 1
│   └── 2
└── b
    ├── 1
    └── 2

2 directories, 4 files
(dvc) jiojiajiu@pon4ik:~/workspace/dvc/list (dvc_list)$ python -m dvc list .
a
b
(dvc) jiojiajiu@pon4ik:~/workspace/dvc/list (dvc_list)$ python -m dvc list . -R
a
a/1
a/2
b
b/1
b/2
```

Reply:
This was multiple target, not recursive case. We are not showing target itself, so files in different targets are mixed, anyway this won't be an issue if we don't support multiple targets.